scene/gui/split_container.h

```diff
@@ -106,24 +122,44 @@ class SplitContainer : public Container {
 
 	Ref<Texture2D> _get_grabber_icon() const;
 	Ref<Texture2D> _get_touch_dragger_icon() const;
-	void _touch_dragger_mouse_exited();
-	void _touch_dragger_gui_input(const Ref<InputEvent> &p_event);
-	void _compute_split_offset(bool p_clamp);
+	Point2i _get_valid_range(int p_dragger_index) const;
+
+	PackedInt32Array _get_desired_sizes() const;
+	void _set_desired_sizes(const PackedInt32Array &p_desired_sizes, int p_priority_index = -1);
+
+	void _update_default_dragger_positions();
+	void _update_dragger_positions(int p_clamp_index = -1);
 	int _get_separation() const;
 	void _resort();
-	Control *_get_sortable_child(int p_idx, SortableVisibilityMode p_visibility_mode = SortableVisibilityMode::VISIBLE_IN_TREE) const;
+	void _update_draggers();
+	void _on_child_visibility_changed(Control *p_control);
+	void _add_valid_child(Control *p_control);
+	void _remove_valid_child(Control *p_control);
 
 protected:
 	bool is_fixed = false;
 
 	void _notification(int p_what);
 	void _validate_property(PropertyInfo &p_property) const;
+
+	virtual void add_child_notify(Node *p_child) override;
+	virtual void remove_child_notify(Node *p_child) override;
+	virtual void move_child_notify(Node *p_child) override;
 	static void _bind_methods();
 
+#ifndef DISABLE_DEPRECATED
+	void _clamp_split_offset_compat_90411();
+	static void _bind_compatibility_methods();
+#endif // DISABLE_DEPRECATED
+
 public:
-	void set_split_offset(int p_offset);
-	int get_split_offset() const;
-	void clamp_split_offset();
+	void set_split_offset(int p_offset, int p_index = 0);
```

Comment:
```suggestion
	void set_split_offset(int p_index, int p_offset);
```

For consistency with other setters in the engine, the index should come first

Reply:
There already is another method for compatibility `_set_split_offset_first` where I do that, this one is for use in the editor. The offset is the most important parameter in this case, and the index is 0 in most cases so I wanted to have it as a default parameter.

Reply:
Okay it doesn't matter that much since it will be rarely used in GDScript.

Reply:
This function isn't even exposed to GDScript.